client-ui/src/app/components/dashboard/dashboard.component.ts

```diff
@@ -13,200 +13,131 @@
+    this.store.dispatch(new ExecutionEventsActions.Get(
+      {
+        asAtTime: this.asAtTime,
+        timestampStart: dateRange && dateRange[0],
+        timestampEnd: dateRange && dateRange[1],
+        offset: dashboardState.pagination.page,
+        sortName: dashboardState.sort.field,
+        sortDirection: dashboardState.sort.direction,
+        searchTerm: dashboardState.filters.searchQuery,
+      }))
+  }
 
-    this.subscriptions.push(
-      this.table.sort.pipe(
-        tap(_ => this.loading = true),
-        map(event => event.sorts[0]),
-      ).subscribe(
-        sort => {
-          this.queryParams = { ...this.queryParams, offset: 0, sortName: sort.prop, sortDirection: sort.dir }
-          this.applyFilters()
-        }
-      )
-    )
+  ngOnDestroy(): void {
+    this.subscriptions.forEach(s => s.unsubscribe())
+  }
 
-    this.subscriptions.push(
-      fromEvent<any>(this.searchInput.nativeElement, 'keyup')
-        .pipe(
-          tap(_ => this.loading = true),
-          map(event => event.target.value),
-          debounceTime(400)
-        ).subscribe(
-          searchTerm => {
-            this.queryParams = { ...this.queryParams, offset: 0, searchTerm: searchTerm }
-            this.applyFilters()
-          }
-        )
-    )
+  onSelect(event): void {
+    const executionEventId = event.selected[0].executionEventId
+    const params = {} as RouterStateUrl
+    params.queryParams = {"executionEventId": executionEventId}
+    params.url = "/app/lineage-overview/"
+    this.store.dispatch(new RouterAction.Go(params))
+  }
 
+  onFilterByDateSwitchToggle(value): void {
+    const dateRange = value
+      ? this.executionEvents.totalDateRange as DateRange
+      : undefined;
+    this.store.dispatch(new DashboardActions.SetDateRange(dateRange))
   }
 
-  private applyFilters(): void {
-    if (this.useDateRange) {
-      this.store.dispatch(new RouterAction.ReplaceUrlState(this.queryParams))
-    }
-    this.store.dispatch(new ExecutionEventsActions.Get({ ...this.queryParams, asAtTime: this.asAtTime }))
-    this.loading = false
+  onDateRangeSelected(range): void {
+    this.store.dispatch(new DashboardActions.SetDateRange(range))
   }
 
-  public onUseDateRangeChange(value): void {
-    this.useDateRange = value
-    this.queryParams = !this.useDateRange && {}
-    this.store.dispatch(new RouterAction.ReplaceUrlState(this.queryParams))
+  onPageChange(page) {
+    this.store.dispatch(new DashboardActions.SetPageNumber(page.offset))
   }
 
-  public onSelect(event): void {
-    const executionEventId = event.selected[0].executionEventId
-    const params = {} as RouterStateUrl
-    params.queryParams = { "executionEventId": executionEventId }
-    params.url = "/app/lineage-overview/"
-    this.store.dispatch(new RouterAction.Go(params))
+  onSortChange(sort: TableSort) {
+    this.store.dispatch(new DashboardActions.SetSortOrder(sort.prop, sort.dir))
   }
 
+  onSearchTermChange = _.debounce(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The class property 'onSearchTermChange' must be marked either 'private', 'public', or 'protected'](https://app.codacy.com/app/ABSA_OSS/spline/pullRequest?prid=4491035)